core/consensus/synchronizer/synchronizer.hpp

```diff
@@ -18,58 +16,14 @@ namespace kagome::consensus {
    * *synchronize* this node's state with the state of other nodes, for example,
    * if this node does not have some blocks
    */
-  struct Synchronizer {
-    virtual ~Synchronizer() = default;
-
-    /**
-     * Start accepting requests for blocks
-     */
-    virtual void start() = 0;
-
+  struct Synchronizer : public network::SyncProtocolClient,
+                        public network::SyncProtocolObserver {
+    // TODO(akvinikym) PRE-304: move it to BabeGossiper
```

Comment:
Is it painful to move it to gossiper in this PR?

Reply:
Yeah, implementation needs to be moved, which will require further changes. Better to do it in the future implementation PR